packages/dashboard/src/components/tasks/utils.ts

```diff
@@ -2,46 +2,66 @@ import { SubmitTask } from 'api-client';
 import * as RmfModels from 'rmf-models';
 
 /* istanbul ignore next */
-function checkField(obj: Record<string, unknown>, field: string, type: string): void {
-  if (!Object.prototype.hasOwnProperty.call(obj, field) || typeof obj[field] !== type) {
-    throw new TypeError(`expected [${field}] to be [${type}]`);
+function checkField(
+  obj: Record<string, unknown>,
+  field: string,
+  type: string,
+  errMsgs: string[],
+): void {
+  try {
+    if (!Object.prototype.hasOwnProperty.call(obj, field) || typeof obj[field] !== type) {
+      throw new TypeError(`expected [${field}] to be [${type}]`);
+    }
+  } catch (err) {
+    errMsgs.push(err.message);
   }
 }
 
+export interface SubmitTaskDetails {
```

Comment:
This shouldn't be needed as `parseTasksFile` should just throw an error on failure (which should be documented). The caller should call this in a try/catch and show an error snackbar if it fails, the error message is available from the thrown error.